### PR TITLE
[Foundation] Bridged dispatch_data_t to NSData to Data should enumerate bytes with the offset not the length parameter as the byte index

### PR DESCRIPTION
**Pull request by phausler**
Resolves https://bugs.swift.org/browse/SR-4028

### CODE REVIEW

**Comment by phausler** [comment]
@swift-ci test

File: test/stdlib/TestData.swift [hunk 1]
```diff
@@ -627,11 +627,15 @@ class TestData : TestDataSuper {
...
 
         var numChunks = 0
+        var offsets = [Data.Index]()
         data.enumerateBytes() { buffer, offset, stop in
             numChunks += 1
+            offsets.append(offset)
         }
 
         expectEqual(2, numChunks, "composing two dispatch_data should enumerate as structural data as 2 chunks")
+        expectEqual(0, offsets[0], "composing two dispatch_data shoudl enumerate as structural data with the first offset as the location of the region")
```
Comment (jrose-apple) [inline test/stdlib/TestData.swift]: Typo: "shoudl"

**Comment by phausler** [comment]
@swift-ci please test

**Comment by phausler** [comment]
@swift-ci test and merge

**Pull request merged**